src/WindowManager.vala

```diff
@@ -795,22 +795,61 @@ namespace Gala {
             return (proxy in modal_stack);
         }
 
+        struct DimData {
+            Meta.Window child;
+            unowned Meta.WindowActor parent_actor;
+        }
+
+        private List<DimData?> dim_data = new List<DimData?> ();
```

Comment:
What about using a `GLib.HashTable<Meta.Window child, unowned Meta.WindowActor>` ?

Reply:
I'm have no experience with `HashTable` in Vala,  what hash and equal functions should it use?

Reply:
GLib.direct_hash and GLib.direct_equal for pointer-based types :)

Reply:
Thanks!